src/modules/configurator/components/UserConfigurations.tsx

```diff
@@ -0,0 +1,54 @@
+import { ConfigureLinkButton } from "./ConfigureLinkButton";
+import { ConfigurationsEmpty } from "./ConfigurationsEmpty";
+import { ConfigurationsSaved } from "./ConfigurationsSaved";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { userConfigurationsState } from "../state";
+import { useEffect, useState } from "react";
+import { userState } from "../../authentification/state";
+import { fetchAllUserConfigurations } from "../services";
+import { PageLoading } from "../../global/components";
+
+export function UserConfigurations() {
+  const [userConfigurations, setUserConfigurations] = useRecoilState(
+    userConfigurationsState
+  );
+  const currentUser = useRecoilValue(userState);
+  const [isDataFetching, setIsDataFetching] = useState(true);
+
+  const handleUserConfigurationsFetch = async (id: string) => {
+    try {
+      const response = await fetchAllUserConfigurations(id);
+      setUserConfigurations(response);
+    } catch (err: any) {
+      throw Error(err);
+    } finally {
+      setIsDataFetching(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!currentUser) return;
+    handleUserConfigurationsFetch(currentUser.id);
+  }, []);
```

Comment:
CuurentUser should be a dependency.

Reply:
Added `currentUser` to dependency array.